src/utils/MetaboliteSpectralMatcher.cpp

```diff
@@ -103,6 +106,8 @@ class TOPPMetaboliteSpectralMatcher :
     setValidFormats_("database", {"mzML", "msp", "mgf"});
     registerOutputFile_("out", "<file>", "", "mzTab file");
     setValidFormats_("out", ListUtils::create<String>("mzTab"));
+    registerOutputFile_("out_merged_spectra", "<file>", "", "Output merged spectra as mzML file if algorithm:merge_spectra is set to true (default).", false);
```

Comment:
What happens if the outputfile is empty but merge_spectra is true?

Keep in mind that the passing of outputfiles is not obvious in UI based workflow managers. By default, an output file is always given. If it is not created by the tool, the workflow will be in trouble.

Reply:
In that case the spectra get merged but no output file is written.
The parameter `out_merged_spectra` is set as optional. Does this get handled as an optional output file like in other TOPP tools?

Reply:
One possibility that is more consistent would be to make an optional "out_spectra"  parameter that gets written in both cases (with/without merging).

Reply:
Yes, if wanting both is rare, this is a much better option

Reply:
Right now it doesn't get written if merging is deactivated. But of course we could write in both cases as suggested. I will comment accordingly in the docs.